waku/v2/node/wakunode2.nim

```diff
@@ -176,16 +176,37 @@ proc stop*(node: WakuNode) {.async.} =
 
   node.started = false
 
+proc subscribe(node: WakuNode, topic: Topic, handler: Option[TopicHandler]) =
+  info "subscribe", topic=topic
+
+  proc defaultHandler(topic: string, data: seq[byte]) {.async, gcsafe.} =
+    # A default handler should be registered for all topics
+    trace "Hit default handler", topic=topic, data=data
+
+    let msg = WakuMessage.init(data)
+    if msg.isOk():
+      node.filters.notify(msg.value(), "")  # Trigger filter handlers on a light node
+      await node.subscriptions.notify(topic, msg.value()) # Trigger subscription handlers on a store/filter node
+      waku_node_messages.inc(labelValues = ["relay"])
+
+  let wakuRelay = node.wakuRelay
+
+  if topic notin PubSub(wakuRelay).topics:
+    # Add default handler only for new topics
+    debug "Registering default handler", topic=topic
+    wakuRelay.subscribe(topic, defaultHandler)
+
+  if handler.isSome:
+    debug "Registering handler", topic=topic
+    wakuRelay.subscribe(topic, handler.get())
+
 proc subscribe*(node: WakuNode, topic: Topic, handler: TopicHandler) =
```

Comment:
Is there a specific reason not to make the Option[TopicHandler] the default public signature?